zrml/prediction-markets/src/lib.rs

```diff
@@ -208,9 +260,28 @@ mod pallet {
                 market.status == MarketStatus::Reported || market.status == MarketStatus::Disputed,
                 Error::<T>::InvalidMarketStatus,
             );
-            Self::clear_auto_resolve(&market_id)?;
+            let (ids_len, disputes_len) = Self::clear_auto_resolve(&market_id)?;
             let market = T::MarketCommons::market(&market_id)?;
-            let weight = Self::on_resolution(&market_id, &market)?;
+            let _ = Self::on_resolution(&market_id, &market)?;
+            let weight;
+            if let MarketType::Categorical(_) = market.market_type {
+                if let MarketStatus::Reported = market.status {
+                    weight =
+                        T::WeightInfo::admin_move_market_to_resolved_categorical_reported(ids_len);
+                } else {
+                    weight = T::WeightInfo::admin_move_market_to_resolved_categorical_disputed(
+                        ids_len,
+                        disputes_len,
+                    );
+                }
+            } else if let MarketStatus::Reported = market.status {
+                weight = T::WeightInfo::admin_move_market_to_resolved_scalar_reported(ids_len);
+            } else {
+                weight = T::WeightInfo::admin_move_market_to_resolved_scalar_disputed(
+                    ids_len,
+                    disputes_len,
+                );
+            }
```

Comment:
[nitpick] This is prone to error. It's not exactly clear that the `else` branch is for disputed scalar markets. When we add other market types, this might be a problem.

I think matching `market.market_type` and `market.status` ~~(and raising an error is the `status` is not as expected!)~~ is safer.

Reply:
2447d4986b4d42182fd4edb88413d2decc69f4bf